src/internal/account.ts

```diff
@@ -404,16 +405,39 @@ export async function getAccountTransactionsCount(args: {
 export async function getAccountCoinAmount(args: {
   aptosConfig: AptosConfig;
   accountAddress: AccountAddressInput;
-  coinType: MoveStructId;
+  coinType?: MoveStructId;
+  faMetadataAddress?: AccountAddressInput;
 }): Promise<number> {
-  const { aptosConfig, accountAddress, coinType } = args;
+  const { aptosConfig, accountAddress, coinType, faMetadataAddress } = args;
+  let faAddress: string;
+
+  if (coinType !== undefined && faMetadataAddress !== undefined) {
+    faAddress = AccountAddress.from(faMetadataAddress).toStringLong();
+  } else if (coinType !== undefined && faMetadataAddress === undefined) {
+    // TODO Move to a separate function as defined in the AIP for coin migration
+    if (args.coinType === APTOS_COIN) {
+      faAddress = AccountAddress.A.toStringLong();
+    } else {
+      faAddress = createObjectAddress(AccountAddress.A, coinType).toStringLong();
```

Comment:
@lightmark to check that this function is correct

Reply:
I took this from the code directly here, which should be extracted to a view function:

https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L320-L328

Reply:
@rtso what is the `asset_type` is not changed from `0x1::aptos_coin::AptosCoin` to `0xa` yet?
From the client perspective, they don't know what to use. Although may be a short term issue.